Python/ibmcloudsql/catalog_table.py

```diff
@@ -0,0 +1,311 @@
+import time
+import logging
+logger = logging.getLogger(__name__)
+
+class HiveMetastore():
+    def __init__(self, target_url):
+        self.current_catalog_table_name = None
+        # keep tracks of what catalog tables are availables
+        self.partitioned_tables = set()
+        self.regular_tables = set()
+        self.sql_stmt_show_temmplate = """
+        SHOW TABLES {like}
+        INTO {cos_out} STORED AS CSV
+        """
+        # The default URL where data should be queried
+        self.cos_in_url_partitioned = "cos://us-geo/sql/customers_partitioned.csv"
+        self.cos_in_url = "cos://us-geo/sql/customers.csv"
+
+        self.sql_stmt_create_template = """
+        CREATE TABLE {table_name}
+        USING CSV
+        LOCATION {cos_in}
+        """
+        self.sql_stmt_show_temmplate = """
+        SHOW TABLES
+        INTO {cos_out} STORED AS CSV
+        """
+        self._target_url = target_url
+
+    @property
+    def target_url(self):
+        return self._target_url
+    @target_url.setter
+    def target_url(self, target_url):
+        self._target_url = target_url
+
+    # Extended functionality
+    def show_catalog_tables(self, tartget_cos_url=None, pattern=None):
```

Comment:
I don't think the "_catalog_" element in these method names is needed. Just name it show_table, drop_table etc. in the module.

Reply:
Updated in commit `00fc7b5bd48668386d01dad60f89b1069c8f8f96`